### DISCUSSION TITLE
Join Field with filterOption

### DISCUSSION THREAD

**Discussion by michalklor** [payloadcms/payload#8937] [Feature Requests & Ideas]
**Description:**

When using the filterOptions function to filter data in collections that include join fields, I've encountered a limitation where the joins parameter is not supported. This restriction prevent our ability to filter documents, such as media, based on their association with other documents, like a specific post.

**Current Behavior:**

Currently, we can only use the where clause for filtering, which significantly limits the functionality of our applications that relies on join fields for filtering options.

**Expected Behavior:**

I would like to propose the addition of join fields support within the filterOptions function. Since the join fields functionality is already available in the REST API of Payload CMS, it would be beneficial to have it implemented in filterOptions as well. This enhancement would allow us to filter data more effectively with relationships between documents.

**Use Case:**

For example, filtering media documents based on their association with a specific post would enhance the user experience and data management capabilities.

**Comment by DanRibbens** [comment]
Hi @michalklor,

Can you give me an example configuration that I can toy with? I think what you're asking for should be possible.

I was just experimenting with the following and it does work:

```ts
{
  collections: [{
  slug: 'categories',
  fields: [
    {
      name: 'name',
      type: 'text',
    },
    {
      name: 'posts',
      type: 'join',
      collection: 'posts',
      on: 'category',
    },
    {
      name: 'isFiltered',
      type: 'checkbox',
      defaultValue: false,
    },
  ],
  }, {
    slug: 'posts',
    fields: [
         {
	      name: 'title',
	      type: 'text',
   		 },
	    {
	      name: 'category',
	      type: 'relationship',
	      relationTo: categoriesSlug,
	      filterOptions: {
	        isFiltered: {
	          equals: false,
	        },
	      },
	    },
      ],
 }
]
// etc...
```

The only awkward thing to note is that I did have to save the category and even if I said this category should be filtered using the checkbox, the UI still let's you click Add Post. It then errors because the prefilled category isn't allowed to be used.

**Reply by DanRibbens** [reply]
I believe I misunderstood your request. In the above config example, I had filterOptions on the relationship.

It seems that maybe you're asking for `filterOptions` on the `join` field. If so this is the same as https://github.com/payloadcms/payload/discussions/8936.

**Comment by DanRibbens** [comment]
https://github.com/payloadcms/payload/pull/8973
